docs/serverdev/index.rst

```diff
@@ -0,0 +1,130 @@
+Contributing to Sentry
+======================
+
+This part of the documentation guides you towards contributing to `Sentry <https://www.github.com/getsentry/sentry>`_.
+It covers the installation and configuration of required dependencies, as well as
+standard procedures of contribution.
+
+Setting up Your Development Environment
+---------------------------------------
+
+Macintosh OSX
+'''''''''''''
+
+Third Party Services
+~~~~~~~~~~~~~~~~~~~~
+To run the basic development server, you need to install both Redis and Postgresql.
+Below are basic commands to install them on OSX. By now, you hopefully have Homebrew installed.
+If you don't run::
+
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+.. note:: If you would like to have the following services run when you start your machine,
+  follow the instructions that Homeberw provides at the end of running ``install``.
+
+Redis
+*****
+Redis is a popular in-memory datastore that Sentry uses for queuing and caching mechanisms.
+Run the following to install, configure, and execute Redis as a daemonized server::
+
+    brew install redis
+    redis-server --save --daemonize yes
+
+Postgresql
+**********
+Postgresql is the primary database that Sentry uses for all persistent storage.
+Run the following to install, configure, and execute Postgresql as a daemonized server::
+
+    brew install postgresql
+    pg_ctl -D /usr/local/var/postgres start
+
+Third Party Libraries
+~~~~~~~~~~~~~~~~~~~~~
+Sentry depends on a combination of both Python and JavaScript packages that need to be installed
+in order for the server to function properly. Below are basic commands to install the languages
+and their libraries on OSX.
+
+Python
+******
+While OSX ships with an acceptable version of Python, installing the latest binaries from HomeBrew
+is recommended. In addition, it is highly recommended to separate your libraries per-environment
+with the use of virtual environments. One of the easiest ways to curate your virtual environments is
+with `virtualenvwrapper <https://virtualenvwrapper.readthedocs.org/>`_. Run the following to install
+and configure brewed Python and virtualenvwrapper::
+
+    brew install python
+    pip install virtualenvwrapper
```

Comment:
For people who are new to python, I generally point at [virtualenv-burrito](https://github.com/brainsik/virtualenv-burrito) which I'm a core contributor to. It just packages up virtualenv + virtualenvwrapper which is nice.